### ISSUE TITLE
Images not loaded due to security headers restrictions

### ISSUE THREAD

**Issue by zainfathoni** [kawalcovid19/wargabantuwarga.com#716]
## Description

<img width="838" alt="Screenshot 2021-08-14 at 09 46 20" src="https://user-images.githubusercontent.com/6315466/129430862-14d48723-8b09-4eaf-a237-2b6c8c28e40f.png">

<img width="602" alt="Screenshot 2021-08-14 at 09 45 48" src="https://user-images.githubusercontent.com/6315466/129430866-faa81529-9302-4e1b-a015-8b47aa828617.png">

## Expected Behaviour

Images are loaded

**Comment by zainfathoni** [comment]
Is it related to your PWA implementation in #705 combined with our new security policies, Mas @mazipan?

<img width="854" alt="Screenshot 2021-08-14 at 10 02 41" src="https://user-images.githubusercontent.com/6315466/129431198-daa6c308-1ff7-4ff1-8a18-b905a094630a.png">

**Comment by zainfathoni** [comment]
I'm also trying another approach by adding `res.cloudinary.com` to the content security policy header on #719 PR.

**Comment by zainfathoni** [comment]
Let's reopen this issue until it's confirmed to be resolved.

**Comment by resir014** [comment]
I think this is due to service worker. The images work just fine before.

**Comment by zainfathoni** [comment]
Apparently so, I tried accessing the deploy preview on https://deploy-preview-719--wargabantuwarga.netlify.app/ once the PWA kicks in, it's failing again.

https://deploy-preview-719--wargabantuwarga.netlify.app/

**Comment by zainfathoni** [comment]
Let's merge #718, then.

@resir014, do you think we need to revert #719, since the changes are irrelevant. 😅 🙏

**Comment by mazipan** [comment]
Revert it first, I will recheck the implementation later when have time.

**Comment by zainfathoni** [comment]
@resir014 I'm reverting #719 at #720.

**Comment by resir014** [comment]
Shall we remove the service worker first until we can figure out a way to add it without messing up service worker?

Since we already added service worker, we should also modify our service worker to become a self-destructing service worker, just in case people already have it installed:

```js
self.addEventListener('install', () => {
  self.skipWaiting();
});

self.addEventListener('activate', () => {
  self.registration.unregister()
    .then(() => {
      return self.clients.matchAll();
    })
    .then((clients) => {
      clients.forEach(client => client.navigate(client.url))
    });
});
```

**Comment by zainfathoni** [comment]
Mas @mazipan, where did the workbox added the `service-worker.js` file?

According to [this guide](https://github.com/NekR/self-destroying-sw),

We need to replace the file content with that code snippet so the browser would uninstall the previously installed service worker.

**Comment by zainfathoni** [comment]
I'm still experiencing the issue on my local since the service worker is still kicking in.

**Comment by ekafyi** [comment]
I'm outside and accessing from my phone (so can't investigate dev console) but +1 having the same behaviour as @zainfathoni on https://deploy-preview-719--wargabantuwarga.netlify.app/

- initially images load fine
- on 2nd/3rd reload once the service worker has taken control (as indicated by "add to home screen" menu) images don't load

**Comment by mazipan** [comment]
It's generated by next-pwa.

**Comment by zainfathoni** [comment]
#721 is my attempt to fix the stale service worker issue, CMIIW @resir014 @mazipan.

**Comment by zainfathoni** [comment]
The bug is fixed now, the service worker is automatically cleared and the image can be loaded.

Thanks for your assistance, @resir014 & @mazipan. 😁 🙏

**Comment by ekafyi** [comment]
Relevant resources for further investigation in the future.

Next-pwa uses Workbox under the hood, so this may be relevant: https://www.npmjs.com/package/cloudinary-workbox-plugin

More on SW cross origin caching behaviour:
- https://whatwebcando.today/articles/opaque-responses-service-worker/
- https://stackoverflow.com/questions/54619653/can-a-service-worker-fetch-and-cache-cross-origin-assets

**Comment by svspicious** [comment]
I am new with CSP stuff, but I think this might be a clue https://qubyte.codes/blog/content-security-policy-and-service-workers.

TLDR; SW is not using `image-src` policy but using `connect-src` policy instead. Since we don't have `connect-src` it fallback to `default-src` wich is only limited to `self` / origin of the domain itself. So, it refuses to **_"connect"_**

![image](https://user-images.githubusercontent.com/35674157/129854554-dae2d309-90d3-4c49-bf32-bd402dc1f8bd.png)

Mas @zainfathoni tried to add Cloudinary url to `script-src` (at #719 ) but ofc it doesn't work, we need to add to `connect-src`.

So I tried to add res.cloudinary.com to `connect-src` and deploy to my personal netlify account for preview (yes, I'm dumb. brb making a Draft PR for deploy preview). And it works (and it cache cloudinary img assets)

https://hardcore-noyce-3b9626.netlify.app/

![image](https://user-images.githubusercontent.com/35674157/129855347-596db943-77f7-4e81-94aa-48cc0ba767b4.png)
![image](https://user-images.githubusercontent.com/35674157/129855773-d3fbd663-728b-477c-8797-2e7fc5970199.png)
